MapboxNavigation/RouteMapViewController.swift

```diff
@@ -490,9 +518,97 @@ class RouteMapViewController: UIViewController {
             self.laneViewsContainerView.isHidden = true
         }, completion: nil)
     }
+    //MARK: End Of Route
+    
+    func showEndOfRoute(duration: TimeInterval = 0.3, completion: ((Bool) -> Void)? = nil) {
+        view.layoutIfNeeded() //flush layout queue
+        
+        endOfRouteContainer.isHidden = false
+        endOfRouteHide.isActive = false
+        endOfRouteShow.isActive = true
+        bannerHide.isActive = true
+        bannerShow.isActive = false
+        bannerContainerShow.isActive = false
+       
+        
+        mapView.enableFrameByFrameCourseViewTracking(for: duration)
+        mapView.setNeedsUpdateConstraints()
+        
+        let animate = {
+            self.view.layoutIfNeeded()
+            self.buttonStack.alpha = 0.0
+        }
+        
+        let noAnimation = { animate(); completion?(true) }
+
+        guard duration > 0.0 else { return noAnimation() }
+        UIView.animate(withDuration: duration, delay: 0.0, options: [.curveLinear], animations: animate, completion: completion)
+        
+        // Prevent the user puck from floating around.
+        if let location = routeController.location {
+            mapView.updateCourseTracking(location: location, animated: false)
```

Comment:
Nit: `updateCourseTracking(location:animated)` takes an optional location so no need to put this in an if let.

Reply:
- [ ] remove redundant optional